### DISCUSSION TITLE
Kelvin, when is the boring docs coming?

### DISCUSSION THREAD

**Discussion by DominusKelvin** [sailscastshq/boring-stack#76] [Ideas]
Hey all, let me know what pains you've currently seen when using The Boring JavaScript Stack, I plan on having a version of the docs available by the end of this week so let me know what you want to see covered.

**Comment by lennyAiko** [comment]
One thing to keep in mind is covering essential features like Link, Head and useForm (amongst others) that are covered in the inertia docs.
Just leaving it here so you can be remember.

Another thing is covering CRUD operations.

**Reply by DominusKelvin** [reply]
So for guides instead of Links and Head, I'll use terms like navigation in the Guide section. Does that make sense @lennyAiko

**Reply by lennyAiko** [reply]
Yes `Navigation` makes sense, it is clear enough to be differentiated from `Routing`

**Reply by DominusKelvin** [reply]
Yes there will also be a guide for routing which will link to the Sails docs for more extensive deep dive. Basically The Boring JavaScript Stack docs will provide enough details for you to quickly glance and get on your way building your app but if you want to go deeper, then you can click through to the extensive docs in the technology you are interested in.

**Reply by lennyAiko** [reply]
That's good enough for me!

**Comment by lennyAiko** [comment]
One thing I’ll also like to know is how you plan on covering features on sails?
Since a lot of it are already on the docs

**Reply by DominusKelvin** [reply]
Yeah should be fun to figure all these out.

**Comment by DominusKelvin** [comment]
Okay I have started cooking...

https://x.com/Dominus_Kelvin/status/1752039403419992097?s=20

**Comment by DominusKelvin** [comment]
Also we are tracking this with #77 and also there is a draft PR on the Sailscasts Docs https://github.com/sailscastshq/docs.sailscasts.com/pull/22

**Comment by lennyAiko** [comment]
Another thing to probably add to the docs is configuring types.

I have been getting a lot of type errors on the sails part of TBJS. The syntax is correct but the type errors just wont give up. so I have to manually disable checking on each file I notice the error.

**Reply by DominusKelvin** [reply]
> Another thing to probably add to the docs is configuring types.
> 
> 
> 
> I have been getting a lot of type errors on the sails part of TBJS. The syntax is correct but the type errors just wont give up. so I have to manually disable checking on each file I notice the error.

Oh this is true. What Syntax is TypeScript complaining about?

**Reply by lennyAiko** [reply]
Take for example I try to query my database:

let category = _Category_.find({})

So I get a type error unless the model is imported into the action (which should not be done)

Do let me know if you understand.

(imagine the italic word is a red line.)

**Reply by DominusKelvin** [reply]
> Take for example I try to query my database:
> 
> 
> 
> let category = _Category_.find({})
> 
> 
> 
> So I get a type error unless the model is imported into the action (which should not be done)
> 
> 
> 
> Do let me know if you understand.
> 
> 
> 
> (imagine the italic word is a red line.)

Yeah I know what's up here. If you check your codebase you'd find a types/index.d.ts where you'd manually define models types. I think we might need Codegen for this or just take it out for the template everyone have. It's a bit tricky.

But do note it's not an error it's just TypeScript type checker warning

**Reply by lennyAiko** [reply]
Yeah I figured it was just a warning. But I just like my screen looking clean. I will check out the `types/index.d.ts`. Thanks

**Reply by DominusKelvin** [reply]
> Yeah I figured it was just a warning. But I just like my screen looking clean. I will check out the `types/index.d.ts`. Thanks

Yeah what you can do to turn it off is in jsconfig.json change checkJS: false and it should be clean ☺️

**Reply by lennyAiko** [reply]
Just the way I want! Thanks

**Reply by DominusKelvin** [reply]
Yeah I should probably document that. Where would you add that in the docs? Just to turn off types if you don't want it in JS files. @lennyAiko

**Reply by lennyAiko** [reply]
It can be under configuration, maybe tag it as optional

**Comment by lennyAiko** [comment]
The documentation should also cover how to redirect for PUT, PATCH, or DELETE requests

**Comment by DominusKelvin** [comment]
A basic version of the docs is now live @lennyAiko 
https://docs.sailscasts.com/boring-stack/getting-started

**Reply by lennyAiko** [reply]
Amazing stuff, you superbly carried out a daunting task!

**Reply by DominusKelvin** [reply]
> Amazing stuff, you superbly carried out a daunting task!

I tried!
